app/frontend/src/features/connections/friends/useRespondToFriendRequest.ts

```diff
@@ -30,9 +31,10 @@ export default function useRespondToFriendRequest() {
         setMutationError("");
         await queryClient.cancelQueries(friendRequestKey("received"));
       },
-      onSuccess: () => {
+      onSuccess: (_, { friendRequest }) => {
         queryClient.invalidateQueries("friendIds");
```

Comment:
Do an optimistic update of the `user` in the query cache here as before. That will reduce buttons jumping around.